common/lib/xmodule/xmodule/js/src/video/03_video_player.js

```diff
@@ -822,14 +815,18 @@ function (HTML5Video, Resizer) {
 
     function updatePlayTime(time, skip_seek) {
         var videoPlayer = this.videoPlayer,
-            duration = this.videoPlayer.duration(),
+            endTime = this.videoPlayer.duration(),
             youTubeId;
 
+        if (this.config.endTime !== null) {
```

Comment:
Why not just say `if (this.config.endTime)` and use Javascript's concept of truthiness? `null` is a falsy value, but so are `false` and `undefined`, and if we have one of those values for `endTime`, this code won't behave the way you expect.